crates/spirv-std/src/ray_tracing.rs

```diff
@@ -159,3 +159,473 @@ bitflags::bitflags! {
         const SKIP_AABBS = 512;
     }
 }
+
+/// A ray query type which is an opaque object representing a ray traversal.
+#[spirv(ray_query)]
+#[derive(Clone, Default)]
+pub struct RayQuery {
+    _private: u32,
+}
+
+/// Constructs an uninitialized ray query variable. Using the syntax
+/// `let (mut)? <name>`. Where `name` is the name of the ray query variable.
+#[macro_export]
+macro_rules! ray_query {
+    (let $name:ident) => {
+        $crate::ray_query!(@inner $name)
+    };
+    (let mut $name:ident) => {
+        $crate::ray_query!(@inner $name, mut)
+    };
+    (@inner $name:ident $(, $mut:tt)?) => {
+        #[allow(deprecated)]
+        let mut $name : *mut RayQuery = core::mem::uninitialized();
```

Comment:
You don't need to make it mutable nor initialize it - you're initializing it through the `asm!`.